src/components/wrapper/AuthWrapper.tsx

```diff
@@ -19,13 +22,22 @@ type Props = {
   export default OrderDetail;
  */
 
-export const AuthWrapper = ({ children }: Props): JSX.Element => {
-  const { status: sessionStatus } = useSession({ required: true });
+export const AuthWrapper = ({ children, maintainerOnly }: Props): JSX.Element => {
+  const { status: sessionStatus, data: session } = useSession({ required: true });
   const authorized = sessionStatus === "authenticated";
+  const isMaintainer = session?.user.role === "MAINTAINER";
   const loading = sessionStatus === "loading";
+  const router = useRouter();
+
+  useEffect(() => {
+    if (session && maintainerOnly && !isMaintainer) {
+      toast.error("Maintainer only route");
+      void router.back();

```

Comment:
<optional>
in general its not advisable to route user back when he hits an aunauthorized page. 

This is chatgpt's explanation:
The recommended user flow for handling access to an authorized page in a web application often involves displaying an unauthorized or access-denied page rather than routing the user back. Here's why:
- Clarity and Transparency: Displaying an unauthorized page provides clear feedback to the user that they do not have permission to access the requested page. It avoids confusion and ambiguity.
- Security: By showing an access-denied page, you maintain the security of your application. It ensures that unauthorized users cannot navigate or guess their way into sensitive areas of the application.
- User Experience: When a user is redirected back to a previous page (e.g., the homepage) without a clear message, they might wonder why they were redirected and how to proceed. An access-denied page provides context and guidance, improving the user experience.
- Error Handling: Access-denied pages often include informative messages or instructions, which can guide the user on what to do next. It can inform them about the need to log in, upgrade their account, or contact support, depending on the specific reason for denial.

Reply:
Switched to user denied page

But we are routing the user when they are not logged in tho